### ISSUE TITLE
Allow subdir to work with variables

### ISSUE THREAD

**Issue by samoht** [ocaml/dune#3385]
It would be nice if the following could work:

```
(subdir %{ocaml_version} ...
```

I haven't found a workaround for this yet, the following didn't work either:

```
(rule (targets src) (action (system "ln -s %{ocaml_version} src")))
(subdir src ...)
```

As `dune` complains that `src` is both generated and static.

**Comment by diml** [comment]
We definitely need better support for projects that need to work with several OCaml versions. Not sure if allowing `subdir` to work with variables is the best approach, but agreed on the motivation. Did you look at merlin BTW? I know they do similar things.

**Comment by samoht** [comment]
I've workaround this using `(rule (copy_files %{ocaml_version}/...)` but it's a bit painful because of #3387  (and also that targets and source_tree deps do not work very well together but I don't have a small repro for this yet)

**Comment by diml** [comment]
Indeed. BTW, for the original idea one problem with going in that direction is that it is quite likely that in practice people won't have one directory per version, but rather one directory per range of versions.
